### DISCUSSION TITLE
Fluxv2: flux-system GitRepository won't reconcile due syntax error, now corrected but still stuck

### DISCUSSION THREAD

**Discussion by dafstone** [fluxcd/flux2#2190] [Q&A] [answered]
The problem:

- We attempted, foolishly to push a change to a GitRepository (not flux) that changed the interval to '1d' -- an illegal setting.  
- This caused the reconciliation of the flux-system GitRepository to fail
- We pushed a change, correcting this error; **however** the GitRepository remains stuck

We get an error at the top of `flux get all` and in the Flux logs that clearly points to the error:

`✗ v1beta1.GitRepositoryList.Items: []v1beta1.GitRepository: v1beta1.GitRepository.Spec: v1beta1.GitRepositorySpec.Reference: Interval: unmarshalerDecoder: time: unknown unit "d" in duration "1d"`

However, at the same time we can see that the GitRepository has moved past the hash with the bad code it in, but still won't reconcile, and we consistantly get context deadline exceeded errors.

We are looking for a way to:

- Fully reset the source controller to pick up the latest changes on flux-system (we've tried a bootstrap, this just made things worse)
- Force the source controller to get the latest version of our flux-system repo as opposed to the bad version it's still loading
- Another solution?

Any help or thoughts?

**Comment by dafstone** [comment, marked as answer]
Well, answering my own question here -- 

The bad push caused annotations on the GitRepository resources in question to become stuck, and they were failing resolution before the Kustomization step could run, which would have altered them.

Manually editing the gitrepository resources in question via `kubectl`, got them unstuck and allowed the Kustomization to succeed.

**Comment by kingdonb** [comment]
This is definitely a bug.

```
$ flux get sources git
✗ v1beta1.GitRepositoryList.Items: []v1beta1.GitRepository: v1beta1.GitRepository.Spec: v1beta1.GitRepositorySpec.Reference: Interval: unmarshalerDecoder: time: unknown unit "d" in duration "1d", error found in #10 byte of ...|rval":"1d","ref":{"b|..., bigger context ...|pec":{"gitImplementation":"go-git","interval":"1d","ref":{"branch":"staging"},"secretRef":{"name":"f|...
```

The `1d` interval is a valid state, according to the validator that let it into my cluster. But it is an invalid state according to the flux CLI, which can't even fetch sources anymore.

I'm glad you're un-stuck, but if you don't mind I'm going to convert this to an issue, as it's something we will need to fix!
